### DISCUSSION TITLE
Why in `eval "$(pyenv init -)"` as indicated in the README.md?

### DISCUSSION THREAD

**Discussion by nyngwang** [pyenv/pyenv#2261] [Q&A] [answered]
You said that these two commands are needed in the README.md:

> MacOS, if Pyenv is installed with Homebrew:
> 
>     echo 'eval "$(pyenv init --path)"' >> ~/.zprofile
>     echo 'eval "$(pyenv init -)"' >> ~/.zshrc

My question is: Why not just:

```
eval "$(pyenv init --path)" >> ~/.zshrc
```

and what's the purpose of `eval "$(pyenv init -)`? It looks like something half-done when seeing `-`, and I never saw any library/package using __both__ `.zshrc` and `.zprofile`. Could you please explain it?

**Comment by hitblast** [comment, marked as answer]
- The `eval` command is used to execute arguments like a shell command. `eval "$(pyenv init -)` is used to execute pyenv at the start of each session inside your terminal.

- When you put the given command inside the `.zshrc` file, it applies to all interactive shells (such as the terminal inside a VSCode project). On the other hand, `.zprofile` is used for non-interactive shells.
